```
============================= test session starts ==============================
platform linux -- Python 3.10.5, pytest-7.1.1, pluggy-1.0.0
rootdir: /build/source, configfile: pyproject.toml
plugins: aiohttp-0.3.0, freezegun-0.4.2, mock-3.8.1, rerunfailures-10.2, socket-0.5.1, xdist-2.5.0, forked-1.4.0, requests-mock-1.9.3, respx-0.19.2, anyio-3.5.0
gw0 [18] / gw1 [18] / gw2 [18] / gw3 [18]1m
..F.......F.......                                                       [100%]
=================================== FAILURES ===================================
_____________________ test_setup_get_doors_errors[pyloop] ______________________
[gw2] linux -- Python 3.10.5 /nix/store/hjwdfamsr1i4hgw41ldbj062icdyshvw-python3-3.10.5/bin/python3.10

hass = <homeassistant.core.HomeAssistant object at 0x7fffee3e5690>

    async def test_setup_get_doors_errors(hass: HomeAssistant) -> None:
        """Test component setup Get Doors Errors."""
        config_entry = MockConfigEntry(
            domain=DOMAIN,
            data=YAML_CONFIG,
            unique_id="test-id",
        )
        config_entry.add_to_hass(hass)
        with patch(
            "homeassistant.components.aladdin_connect.cover.AladdinConnectClient.login",
            return_value=True,
        ), patch(
            "homeassistant.components.aladdin_connect.cover.AladdinConnectClient.get_doors",
            return_value=None,
        ):
>           assert await hass.config_entries.async_setup(config_entry.entry_id) is True
E           assert False is True

config_entry = <tests.common.MockConfigEntry object at 0x7fffee4e0820>
hass       = <homeassistant.core.HomeAssistant object at 0x7fffee3e5690>

tests/components/aladdin_connect/test_init.py:30: AssertionError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
2022-07-10 10:26:22.597 DEBUG    MainThread asyncio:selector_events.py:54 Using selector: EpollSelector
----------------------------- Captured stderr call -----------------------------
INFO:homeassistant.loader:Loaded aladdin_connect from homeassistant.components.aladdin_connect
INFO:homeassistant.setup:Setting up aladdin_connect
INFO:homeassistant.setup:Setup of domain aladdin_connect took 0.0 seconds
ERROR:homeassistant.config_entries:Error setting up entry Mock Title for aladdin_connect
Traceback (most recent call last):
  File "/build/source/homeassistant/config_entries.py", line 353, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/build/source/homeassistant/components/aladdin_connect/__init__.py", line 26, in async_setup_entry
    acc = AladdinConnectClient(username, password, async_get_clientsession(hass))
TypeError: AladdinConnectClient.__init__() takes 3 positional arguments but 4 were given
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=aladdin_connect>
------------------------------ Captured log call -------------------------------
2022-07-10 10:26:22.603 INFO     SyncWorker_0 homeassistant.loader:loader.py:440 Loaded aladdin_connect from homeassistant.components.aladdin_connect
2022-07-10 10:26:22.603 INFO     MainThread homeassistant.setup:setup.py:204 Setting up aladdin_connect
2022-07-10 10:26:22.604 INFO     MainThread homeassistant.setup:setup.py:252 Setup of domain aladdin_connect took 0.0 seconds
2022-07-10 10:26:22.610 ERROR    MainThread homeassistant.config_entries:config_entries.py:416 Error setting up entry Mock Title for aladdin_connect
Traceback (most recent call last):
  File "/build/source/homeassistant/config_entries.py", line 353, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/build/source/homeassistant/components/aladdin_connect/__init__.py", line 26, in async_setup_entry
    acc = AladdinConnectClient(username, password, async_get_clientsession(hass))
TypeError: AladdinConnectClient.__init__() takes 3 positional arguments but 4 were given
2022-07-10 10:26:22.610 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event component_loaded[L]: component=aladdin_connect>
--------------------------- Captured stderr teardown ---------------------------
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_stop[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_final_write[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_close[L]>
---------------------------- Captured log teardown -----------------------------
2022-07-10 10:26:22.614 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_stop[L]>
2022-07-10 10:26:22.614 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_final_write[L]>
2022-07-10 10:26:22.614 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_close[L]>
____________________ test_setup_component_no_error[pyloop] _____________________
[gw2] linux -- Python 3.10.5 /nix/store/hjwdfamsr1i4hgw41ldbj062icdyshvw-python3-3.10.5/bin/python3.10

hass = <homeassistant.core.HomeAssistant object at 0x7fffeda87850>

    async def test_setup_component_no_error(hass: HomeAssistant) -> None:
        """Test component setup No Error."""
        config_entry = MockConfigEntry(
            domain=DOMAIN,
            data=YAML_CONFIG,
            unique_id="test-id",
        )
        config_entry.add_to_hass(hass)
        with patch(
            "homeassistant.components.aladdin_connect.cover.AladdinConnectClient.login",
            return_value=True,
        ):
    
>           assert await hass.config_entries.async_setup(config_entry.entry_id)
E           assert False

config_entry = <tests.common.MockConfigEntry object at 0x7fffee410820>
hass       = <homeassistant.core.HomeAssistant object at 0x7fffeda87850>

tests/components/aladdin_connect/test_init.py:86: AssertionError
---------------------------- Captured stderr setup -----------------------------
DEBUG:asyncio:Using selector: EpollSelector
------------------------------ Captured log setup ------------------------------
2022-07-10 10:26:22.802 DEBUG    MainThread asyncio:selector_events.py:54 Using selector: EpollSelector
----------------------------- Captured stderr call -----------------------------
INFO:homeassistant.loader:Loaded aladdin_connect from homeassistant.components.aladdin_connect
INFO:homeassistant.setup:Setting up aladdin_connect
INFO:homeassistant.setup:Setup of domain aladdin_connect took 0.0 seconds
ERROR:homeassistant.config_entries:Error setting up entry Mock Title for aladdin_connect
Traceback (most recent call last):
  File "/build/source/homeassistant/config_entries.py", line 353, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/build/source/homeassistant/components/aladdin_connect/__init__.py", line 26, in async_setup_entry
    acc = AladdinConnectClient(username, password, async_get_clientsession(hass))
TypeError: AladdinConnectClient.__init__() takes 3 positional arguments but 4 were given
DEBUG:homeassistant.core:Bus:Handling <Event component_loaded[L]: component=aladdin_connect>
------------------------------ Captured log call -------------------------------
2022-07-10 10:26:22.808 INFO     SyncWorker_0 homeassistant.loader:loader.py:440 Loaded aladdin_connect from homeassistant.components.aladdin_connect
2022-07-10 10:26:22.808 INFO     MainThread homeassistant.setup:setup.py:204 Setting up aladdin_connect
2022-07-10 10:26:22.808 INFO     MainThread homeassistant.setup:setup.py:252 Setup of domain aladdin_connect took 0.0 seconds
2022-07-10 10:26:22.811 ERROR    MainThread homeassistant.config_entries:config_entries.py:416 Error setting up entry Mock Title for aladdin_connect
Traceback (most recent call last):
  File "/build/source/homeassistant/config_entries.py", line 353, in async_setup
    result = await component.async_setup_entry(hass, self)
  File "/build/source/homeassistant/components/aladdin_connect/__init__.py", line 26, in async_setup_entry
    acc = AladdinConnectClient(username, password, async_get_clientsession(hass))
TypeError: AladdinConnectClient.__init__() takes 3 positional arguments but 4 were given
2022-07-10 10:26:22.812 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event component_loaded[L]: component=aladdin_connect>
--------------------------- Captured stderr teardown ---------------------------
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_stop[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_final_write[L]>
DEBUG:homeassistant.core:Bus:Handling <Event homeassistant_close[L]>
---------------------------- Captured log teardown -----------------------------
2022-07-10 10:26:22.814 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_stop[L]>
2022-07-10 10:26:22.814 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_final_write[L]>
2022-07-10 10:26:22.815 DEBUG    MainThread homeassistant.core:core.py:886 Bus:Handling <Event homeassistant_close[L]>
=========================== short test summary info ============================
FAILED tests/components/aladdin_connect/test_init.py::test_setup_get_doors_errors[pyloop]
FAILED tests/components/aladdin_connect/test_init.py::test_setup_component_no_error[pyloop]
========================= 2 failed, 16 passed in 3.74s =========================

--- a/pkgs/development/python-modules/aioaladdinconnect/default.nix
+++ b/pkgs/development/python-modules/aioaladdinconnect/default.nix
@@ -1,22 +1,21 @@
 { lib
 , aiohttp
 , buildPythonPackage
-, fetchFromGitHub
+, fetchPypi
 , pythonOlder
 }:
 
 buildPythonPackage rec {
   pname = "aioaladdinconnect";
-  version = "0.1.20";
+  version = "0.1.21";
   format = "setuptools";
 
   disabled = pythonOlder "3.7";
 
-  src = fetchFromGitHub {
-    owner = "mkmer";
-    repo = "AIOAladdinConnect";
-    rev = version;
-    hash = "sha256-JHGAJx1Hb/NKWZ+y1ACmVUiesYj1VVHnYCcP+XZGADs=";
+  src = fetchPypi {
+    pname = "AIOAladdinConnect";
+    inherit version;
+    hash = "sha256-Zr9QLiQNmphPpAVnFVpoOlVbuWUVtWXIacYKAnth+E4=";
   };
 
   propagatedBuildInputs = [

